src/memory-analyzer/analyze_symbol.cpp

```diff
@@ -247,6 +247,29 @@ exprt gdb_value_extractort::get_pointer_to_member_value(
     UNREACHABLE;
   }
 
+  const auto it = values.find(memory_location);
+  // if the structure we are pointing to does not exists we need to build a
+  // temporary object for it: get the type from symbol table, query gdb for
+  // value, allocate new object for it and then store into assignments
+  if(it == values.end())
+  {
+    const auto symbol_expr = struct_symbol->symbol_expr();
+    const auto zero = zero_initializer(symbol_expr.type(), location, ns);
+    CHECK_RETURN(zero.has_value());
+    const auto val = get_expr_value(symbol_expr, *zero, location);
+
+    symbol_exprt dummy(
+      pointer_typet(symbol_expr.type(), config.ansi_c.pointer_width));
```

Comment:
Use `pointer_type` from `util/c_types.{h,cpp}`, and add a name.

Reply:
Done.